### PR TITLE
dist: add udev rule to enable user hidraw access

### PR DESCRIPTION
**Pull request by HurricanePootis**
On linux, whenever a device is connected to the system, the only user/group allowed to access is `root`. This is a problem, however, for the custom drivers for both the joycons and the pro controller. Therefore, I am making this PR to add a udev ruels in the `dist/` folder. 

This is helpful for both appimage/flaptak users and users who get their controller via a system package manager. For the former group, instead of the user having to go to the discord and either see the pinned post with an outdated method using `plugdev` (which, to my understanding, is a debian thing), the user can download this file from the github repo and install it at `/etc/udev/rules.d/99-yuzu-input.rules`. For the latter group, system packagers can automatically include this rule and have it installed at `/usr/udev/rules.d/99-yuzu-input.rules`

Now, I currently don't have access to any switch controllers as I am using a Wii U controller at the moment, but anyone who does and is on linux, please try it out and give me feedback! Hopefully, this PR will get accepted, and in the upcoming new yuzu wiki, linux users can be referred to this file.

### CODE REVIEW

**Comment by v1993** [comment]
You must prefix it with `72-`, `99-` will not work. See https://github.com/systemd/systemd/issues/4288.

**Comment by HurricanePootis** [comment]
> You must prefix it with `72-`, `99-` will not work. See [systemd/systemd#4288](https://github.com/systemd/systemd/issues/4288).

done

File: dist/72-yuzu-input.rules [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#Allow systemd-login to manage user access to hidraw with this file
+#On most systems, this file should be installed to /etc/udev/rules.d/72-yuzu-input.rules
+#Consult your distro if this is not the case
```
Comment (liamwhite) [inline dist/72-yuzu-input.rules:3]:
```suggestion
# Allow systemd-login to manage user access to hidraw with this file
# On most systems, this file should be installed to /etc/udev/rules.d/72-yuzu-input.rules
# Consult your distro if this is not the case
```

File: dist/72-yuzu-input.rules [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+#Nintendo's Switch Pro Controller and Pro Controller for Wired and Bluetooth
```
Comment (liamwhite) [inline dist/72-yuzu-input.rules:5]:
```suggestion
# Switch Pro Controller (USB/Bluetooth)
```

File: dist/72-yuzu-input.rules [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+KERNEL=="hidraw*", ATTRS{idVendor}=="057e", ATTRS{idProduct}=="2009", MODE="0660", TAG+="uaccess"
+KERNEL=="hidraw*", KERNELS=="*057e:2009*", MODE="0660", TAG+="uaccess"
+#Nintendo's Left Joy-Con for Wired and Bluetooth
+KERNEL=="hidraw*", ATTRS{idVendor}=="057e", ATTRS{idProduct}=="2006", MODE="0600", TAG+="uaccess"
+KERNEL=="hidraw*", KERNELS=="*057e:2006*", MODE="0600", TAG+="uaccess"
```
Comment (liamwhite) [inline dist/72-yuzu-input.rules:10]:
```suggestion
# Joy-Con L (Bluetooth)
KERNEL=="hidraw*", KERNELS=="*057e:2006*", MODE="0600", TAG+="uaccess"
```

File: dist/72-yuzu-input.rules [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+#Nintendo's Right Joy-Con for Wired and Bluetooth
+KERNEL=="hidraw*", ATTRS{idVendor}=="057e", ATTRS{idProduct}=="2007", MODE="0600", TAG+="uaccess"
+KERNEL=="hidraw*", KERNELS=="*057e:2007*", MODE="0600", TAG+="uaccess"
```
Comment (liamwhite) [inline dist/72-yuzu-input.rules:13]:
```suggestion
# Joy-Con R (Bluetooth)
KERNEL=="hidraw*", KERNELS=="*057e:2007*", MODE="0600", TAG+="uaccess"

# Joy-Con Charging Grip (USB)
KERNEL=="hidraw*", ATTRS{idVendor}=="057e", ATTRS{idProduct}=="200e", MODE="0600", TAG+="uaccess"
```

**Comment by HurricanePootis** [comment]
Thanks for the suggestions @liamwhite

File: dist/72-yuzu-input.rules [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Allow systemd-login to manage user access to hidraw with this file
```
Comment (liamwhite) [inline dist/72-yuzu-input.rules:1]: Add SPDX header

Comment (liamwhite) [inline dist/72-yuzu-input.rules:1]:
```suggestion
# Allow systemd-logind to manage user access to hidraw with this file
```

**Review by kojq — CHANGES_REQUESTED** [review]
None of these are functionally necessary, but these are suggestions:

- `*057e:2009*` can be changed to `0005:057e:2009.*` and so on for the remaining lines to be more explicit.

- Spacing out from the next comment as was done with `# Joy-Con Charging Grip (USB)` would make this file more readable.

- `Joy-Con L` can be changed to `Joy-Con (L)` and so on for consistent naming.

- `SUBSYSTEM=="hidraw"` can be included to be more explicit.

**Comment by liamwhite** [comment]
@kojq the names of 057e:2006 and 057e:2007 in the [USB ID repository](http://www.linux-usb.org/usb-ids.html) are Joy-Con L and Joy-Con R -- no parentheses.

**Comment by kojq** [comment]
@liamwhite ah, so we are using that for reference. Also, why is `MODE=` RWX permission different for the Pro Controller?

**Comment by liamwhite** [comment]
That seems like a mistake, it shouldn't need +x.

File: dist/72-yuzu-input.rules [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# SPDX-FileCopyrightText: 2023 yuzu Emulator Project
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Allow systemd-logind to manage user access to hidraw with this file
+# On most systems, this file should be installed to /etc/udev/rules.d/72-yuzu-input.rules
+# Consult your distro if this is not the case
+
+# Switch Pro Controller (USB/Bluetooth)
+KERNEL=="hidraw*", ATTRS{idVendor}=="057e", ATTRS{idProduct}=="2009", MODE="0660", TAG+="uaccess"
+KERNEL=="hidraw*", KERNELS=="*057e:2009*", MODE="0660", TAG+="uaccess"
```
Comment (liamwhite) [inline dist/72-yuzu-input.rules:10]:
```suggestion
KERNEL=="hidraw*", KERNELS=="*057e:2009*", MODE="0660", TAG+="uaccess"

```

File: dist/72-yuzu-input.rules [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+# Joy-Con L (Bluetooth)
+KERNEL=="hidraw*", KERNELS=="*057e:2006*", MODE="0660", TAG+="uaccess"
```
Comment (liamwhite) [inline dist/72-yuzu-input.rules:12]:
```suggestion
KERNEL=="hidraw*", KERNELS=="*057e:2006*", MODE="0660", TAG+="uaccess"

```

**Review by liamwhite — APPROVED** [review]

**Pull request merged**